sdks/python/apache_beam/ml/inference/base_test.py

```diff
@@ -339,6 +375,79 @@ def validate_inference_args(
     third_party_model_handler.batch_elements_kwargs()
     third_party_model_handler.validate_inference_args({})
 
+  def test_run_inference_prediction_result_with_model_id(self):
+    examples = [1, 5, 3, 10]
+    expected = [
+        base.PredictionResult(
+            example=example,
+            inference=example + 1,
+            model_id='fake_model_id_default') for example in examples
+    ]
+    with TestPipeline() as pipeline:
+      pcoll = pipeline | 'start' >> beam.Create(examples)
+      actual = pcoll | base.RunInference(
+          FakeModelHandlerReturnsPredictionResult())
+      assert_that(actual, equal_to(expected), label='assert:inferences')
+
+  @pytest.mark.it_postcommit
+  def test_run_inference_prediction_result_with_side_input(self):
+    test_pipeline = TestPipeline(is_integration_test=True)
```

Comment:
Should we be forcing this to be a streaming pipeline? Without perfect windowing, this won't work on Batch (and it doesn't make much sense), right?

Reply:
Relatedly, one thing that I think this PR is missing is validation that we are only running this in streaming mode. We'll probably need something similar to the `ExternalTransformFinder` to find RunInference with model updates at pipeline compile time - https://github.com/apache/beam/blob/3bf17574c6f18942d1015d5a837af23ae83c4d1a/sdks/python/apache_beam/pipeline.py#L1049 - or we'll need a way to read pipeline options during construction of this transform itself.

Its ok to add that in a separate PR, but we definitely need to do it (and document it)

Reply:
I added `RunInferenceSideInputFinder`, which checks for `_enable_side_input_loading` bool. 

I think it is the simple way to go like you suggested. Thanks. 

if `_enable_side_input_loading` is enables and streaming flag not specified, we fail the pipeline during construction phase.

Reply:
LGTM